### DISCUSSION TITLE
Get positions of images, text, shapes from a pdf?

### DISCUSSION THREAD

**Discussion by Rapid1898-code** [jsvine/pdfplumber#903] [Ask for help with specific PDFs]
Hello - i have the attached pdf - 
And i would like to get the the position-information for the different elements in eg. this form

```
{
"images" : [
{
"image_url" : "data:.",
"height" : 1024,
"width" : 1027,
"positionX" : 0,
"positionY" : 0
}
],
"texts" : [
{
"text" : "Text",
"color" : "#000000",
"font" : "arena",
"height" : 1024,
"width" : 1027,
"positionX" : 0,
"positionY" : 0
}
],
"shapes" : [
{
"svg_path" : "M0 A1 0......",
"height" : 1024,
"width" : 1027,
"positionX" : 0,
"positionY" : 0,
"color" : "#D2D2D2"
}
]
}

```

Is this somehow possible?

[bed.pdf](https://github.com/jsvine/pdfplumber/files/11703757/bed.pdf)

**Comment by jsvine** [comment]
Thanks for your interest in this library, @Rapid1898-code. You can get the position information (and other details) directly from the objects. E.g., each object in the `page.rects`, `page.curves`, `page.lines`, and `page.images` lists will have an `x0`, `x1`, `top`, and `bottom` attribute. To get the "texts" you refer to above, you can use `page.extract_words(...)`. I suggest consulting the ["Objects" section of the README.md documentation](https://github.com/jsvine/pdfplumber#objects) to understand what other properties are available.

**Comment by Rapid1898-code** [comment]
Thanks a lot - i think i understand and that helps!
